runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -4064,7 +4061,18 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
    bool useAotCompilation = false;
    try
       {
-      auto req = stream->read<uint64_t, uint32_t, J9Method *, J9Class*, TR_Hotness, std::string,
+      auto response = stream->readCompileRequest();
+      if (response == JITaaS::J9ServerMessageType::clientTerminate)
+         {
+         clientId = std::get<0>(stream->getRecvData<uint64_t>());
+         throw JITaaS::StreamCancel(response, clientId);
+         }
+      else if (response != JITaaS::J9ServerMessageType::compilationRequest)
```

Comment:
I am thinking that this check should be implemented inside `readCompileRequest()` much like `read()` does. We already have a catch block for `StreamMessageTypeMissmatch`